static/js/publisher/market/screenshots.js

```diff
@@ -0,0 +1,249 @@
+      document.querySelector('.js-delete-screenshot').setAttribute('disabled', 'disabled');
+    } else {
+      document.querySelector('.js-delete-screenshot').removeAttribute('disabled');
+    }
+  };
+
+  render();
+
+  const onScreenshotsChange = function() {
+    addScreenshots(this.files);
+    render();
+  };
+
+  // delegated click handlers
+  document.addEventListener("click", function(event){
+    // Delete screenshot
+    if (event.target.classList.contains('js-delete-screenshot')
+        || event.target.parentNode.classList.contains('js-delete-screenshot')
+    ) {
+      event.preventDefault();
+      let screenshot = state.images.filter(image => image.selected)[0];
+      if (screenshot) {
+        deleteScreenshot(screenshot);
+        selectScreenshot();
+      }
+      render();
+    } else if (event.target.classList.contains('js-fullscreen-screenshot')
+        || (event.target.parentNode && event.target.parentNode.classList.contains('js-fullscreen-screenshot'))
+    ) {
+      event.preventDefault();
+      let screenshot = state.images.filter(image => image.selected)[0];
+
+      // if none is selected pick first screenshot from list
+      if (!screenshot) {
+        screenshot = state.images.filter(image => image.type === 'screenshot')[0];
+      }
+
+      if (screenshot) {
+        lightbox.openLightbox(
+          screenshot.url,
+          state.images.filter(image => image.type === 'screenshot').map(image => image.url)
+        );
+      }
+    } else {
+      // unselect any screenshots when clicked outside of them
+      selectScreenshot();
+    }
+
+    // clicking on [+] add screenshots button
+    if (event.target.classList.contains('js-add-screenshots')
+        || event.target.parentNode.classList.contains('js-add-screenshots')
+    ) {
+      event.preventDefault();
+
+      const input = document.createElement('input');
+      input.type = "file";
+      input.accept = "image/*";
+      input.name="screenshots";
+      input.hidden = "hidden";
+
+      screenshotsToolbarEl.parentNode.appendChild(input);
+      input.addEventListener("change", onScreenshotsChange);
+      input.click();
+    }
+
+    // clicking on screenshot to select it
+    if (event.target.classList.contains('p-screenshot')) {
+      event.preventDefault();
+      selectScreenshot(event.target.src);
+      setTimeout(() => {
+        render();
+      }, 50);
+      return;
+    }
+
+    render();
+  });
+
+  document.addEventListener('dblclick', event => {
+    if (event.target.classList.contains('p-screenshot')) {
+      event.preventDefault();
+      let screenshot = state.images.filter(image => image.selected)[0];
+
+      if (screenshot) {
+        lightbox.openLightbox(
+          screenshot.url,
+          state.images.filter(image => image.type === 'screenshot').map(image => image.url)
+        );
+      }
+    }
+  });
+}
+
+export {
+  initSnapScreenshotsEdit,
+  // for testing
+  templates,
```

Comment:
If we need to export things that aren't 'public' api's for a module, should we namespace them? It should provide a red flag to anyone who wants to import that function to maybe move it somewhere more generic.

Something like
```
const __tests = {
  templates: templates,
  serializeState: serializeState
};

export {
  initSnapScreenshotsEdit,
  __tests
}
```

This might be overkill - just a thought.

Reply:
Yes, I was thinking about something like this. Didn't want to over-engineer it at this stage.
In theory you should only need to test public functions (private should be covered if they are used internally), but in this case it's just easier to unit test some helper function then whole thing operating on DOM.

I would leave it for now. Hopefully when we move to React and per field editing it will be better structured and easier to test.